sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.netstandard2.0.cs

```diff
@@ -115,52 +115,90 @@ public static partial class BinaryDataExtensions
-        public static bool operator !=(Azure.Core.Dynamic.JsonData? left, double right) { throw null; }
-        public static bool operator !=(Azure.Core.Dynamic.JsonData? left, int right) { throw null; }
-        public static bool operator !=(Azure.Core.Dynamic.JsonData? left, long right) { throw null; }
-        public static bool operator !=(Azure.Core.Dynamic.JsonData? left, float right) { throw null; }
-        public static bool operator !=(Azure.Core.Dynamic.JsonData? left, string? right) { throw null; }
-        public static bool operator !=(bool left, Azure.Core.Dynamic.JsonData? right) { throw null; }
-        public static bool operator !=(double left, Azure.Core.Dynamic.JsonData? right) { throw null; }
-        public static bool operator !=(int left, Azure.Core.Dynamic.JsonData? right) { throw null; }
-        public static bool operator !=(long left, Azure.Core.Dynamic.JsonData? right) { throw null; }
-        public static bool operator !=(float left, Azure.Core.Dynamic.JsonData? right) { throw null; }
-        public static bool operator !=(string? left, Azure.Core.Dynamic.JsonData? right) { throw null; }
+        internal DynamicJson() { }
+        public static implicit operator bool (Azure.Core.Dynamic.DynamicJson value) { throw null; }
+        public static implicit operator double (Azure.Core.Dynamic.DynamicJson value) { throw null; }
+        public static implicit operator int (Azure.Core.Dynamic.DynamicJson value) { throw null; }
+        public static implicit operator long (Azure.Core.Dynamic.DynamicJson value) { throw null; }
+        public static implicit operator bool? (Azure.Core.Dynamic.DynamicJson value) { throw null; }
+        public static implicit operator double? (Azure.Core.Dynamic.DynamicJson value) { throw null; }
+        public static implicit operator int? (Azure.Core.Dynamic.DynamicJson value) { throw null; }
+        public static implicit operator long? (Azure.Core.Dynamic.DynamicJson value) { throw null; }
+        public static implicit operator float? (Azure.Core.Dynamic.DynamicJson value) { throw null; }
+        public static implicit operator float (Azure.Core.Dynamic.DynamicJson value) { throw null; }
+        public static implicit operator string (Azure.Core.Dynamic.DynamicJson value) { throw null; }
         System.Dynamic.DynamicMetaObject System.Dynamic.IDynamicMetaObjectProvider.GetMetaObject(System.Linq.Expressions.Expression parameter) { throw null; }
         public override string ToString() { throw null; }
+        [System.Diagnostics.DebuggerDisplayAttribute("{Current,nq}")]
+        [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+        public partial struct ArrayEnumerator : System.Collections.Generic.IEnumerable<Azure.Core.Dynamic.DynamicJson>, System.Collections.Generic.IEnumerator<Azure.Core.Dynamic.DynamicJson>, System.Collections.IEnumerable, System.Collections.IEnumerator, System.IDisposable
```

Comment:
I will look into whether this can be made internal and accessed through the dynamic layer.

Reply:
If we are planning to ship the MutableJsonDocument publicly, then this enumerator adds value (as public). It will allow enumerating arrays without the need to allocate (box) the enumerator

Reply:
I agree it makes sense to have MutableJsonElement.ArrayEnumerator as a public type.  Do you feel strongly about also exposing DynamicJson.ArrayEnumerator as a public type?

Reply:
Tracking with https://github.com/Azure/azure-sdk-for-net/issues/33856